src/game.rs

```diff
@@ -33,26 +33,52 @@ impl Game {
     /// After loading the player profile and initializing the current
     /// level, the game consists of repeatedly calling `play_turn`
     /// on the player's `Player` instance.
-    pub fn play(player: impl Player + Send + Sync + 'static) {
+    pub fn play(player: impl Player + Send + Sync + std::clone::Clone + 'static) {
         let mut game = Game::new();
         game.start(player);
     }
 
-    fn start(&mut self, player: impl Player + Send + Sync + 'static) {
+    fn start(&mut self, player: impl Player + Send + Sync + Clone + 'static) {
         let level;
-        if self.profile.maximus_oxidus {
+        if self.profile.challenge_mode {
+            level = self.profile.level - 98;
+            if level == 10 {
+                println!("CONGRATULATIONS! You have compeleted Challenge mode!");
+                self.profile.challenge_mode = false;
+                self.profile.level = 9;
+                // Set back to level 9
+                starter::write_profile(&self.profile, None);
+                println!("Restart if you would like to play again.");
+                return;
+            }
+        } else if self.profile.maximus_oxidus {
             println!("Now that you have earned the title Maximus Oxidus, you may choose to hone your skills on any level.");
+            println!("Or choose 99 for Challenge Mode to traverse all levels!");
             level = ui::select_level();
             starter::write_readme(&self.profile, level, None);
             println!("See (updated) README.md for level {} instructions.", level);
         } else {
             level = self.profile.level;
         }
         println!("Starting Level {}", level);
+        if level == 99 {
+            self.profile.challenge_mode = true;
```

Comment:
I don't think we need to set `challenge_mode` in the player profile. As mentioned in my other comment, choosing to do challenge mode can just be a prompt.